### PR TITLE
Getting display's physical dpi

### PR DESCRIPTION
**Pull request by Rsedaikin**

### CODE REVIEW

File: skiko/build.gradle.kts [hunk 1]
```diff
@@ -915,6 +916,7 @@ fun createLinkJvmBindings(
                     "-static-libgcc",
                     "-lGL",
                     "-lX11",
+                    "-lXrandr",
```
Comment (olonho) [inline skiko/build.gradle.kts:919]: Please use dynamically loaded Xrandr, as in https://github.com/JetBrains/skiko/blob/4fb8aac1b4c277bdc4d1b14095f8fc5583f19f88/skiko/src/jvmMain/cpp/linux/display.cc#L22

File: skiko/src/jvmMain/objectiveC/macos/Drawlayer.mm [hunk 2]
```diff
@@ -399,6 +399,18 @@ JNIEXPORT jint JNICALL Java_org_jetbrains_skiko_SystemTheme_1awtKt_getCurrentSys
...
 
+JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_HardwareLayer_getCurrentDPI(JNIEnv *env, jobject component, jlong platformInfoPtr)
+{
+    @autoreleasepool {
+        LayerHandler *layer = findByObject(env, component);
+        NSScreen *screen = [layer.window screen];
+        NSDictionary *description = [screen deviceDescription];
+        NSSize displayPixelSize = [[description objectForKey:NSDeviceSize] sizeValue];
+        CGSize displayPhysicalSize = CGDisplayScreenSize([[description objectForKey:@"NSScreenNumber"] unsignedIntValue]);
+        return (jlong)(displayPixelSize.width / (displayPhysicalSize.width / 25.4));
```
Comment (olonho) [inline skiko/src/jvmMain/objectiveC/macos/Drawlayer.mm:410]: Could you please explain `25.4`.

File: skiko/src/jvmMain/cpp/windows/drawlayer.cc [hunk 3]
```diff
@@ -57,4 +58,19 @@ extern "C"
                 return 2;
          }
     }
+
+    JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_HardwareLayer_getCurrentDPI(JNIEnv *env, jobject canvas, jlong platformInfoPtr)
+    {
+        HWND hwnd = (HWND)Java_org_jetbrains_skiko_HardwareLayer_getWindowHandle(env, canvas, platformInfoPtr);
+        HMONITOR display = MonitorFromWindow(hwnd, MONITOR_DEFAULTTONEAREST);
+        UINT xDpi, yDpi;
```
Comment (olonho) [inline skiko/src/jvmMain/cpp/windows/drawlayer.cc:66]: xDpi = 0, yDpi = 0

File: skiko/src/awtMain/kotlin/org/jetbrains/skiko/SkiaLayer.awt.kt [hunk 4]
```diff
@@ -140,6 +140,9 @@ actual open class SkiaLayer internal constructor(
 
     val windowHandle: Long
         get() = backedLayer.windowHandle
+    
+    val currentDPI: Long
```
Comment (olonho) [inline skiko/src/awtMain/kotlin/org/jetbrains/skiko/SkiaLayer.awt.kt:144]: Maybe Int?

File: skiko/src/jvmMain/objectiveC/macos/Drawlayer.mm (hunk 2 above)
Comment (Rsedaikin) [reply skiko/src/jvmMain/objectiveC/macos/Drawlayer.mm:410]: dpi - dots per inch
the physical size of the display is in millimeters so we need to convert it to inches so we divide it by 25.4.

Comment (olonho) [reply skiko/src/jvmMain/objectiveC/macos/Drawlayer.mm:410]: For me on 27 inch iMac (height 20.3 inches  width of 25.6 inches ) at 5120 x 2880 this function gives 108 dot per inch, which almost matches claimed 110 dpi.

File: skiko/src/awtMain/kotlin/org/jetbrains/skiko/SkiaLayer.awt.kt (hunk 4 above)
Comment (olonho) [inline skiko/src/awtMain/kotlin/org/jetbrains/skiko/SkiaLayer.awt.kt:144]: Please add docs for new public fields and methods (and documenting nearby fields, unless documented in expect declaration) would be neat.

File: skiko/build.gradle.kts (hunk 1 above)
Comment (Rsedaikin) [reply skiko/build.gradle.kts]: Done.

File: skiko/src/jvmMain/cpp/windows/drawlayer.cc (hunk 3 above)
Comment (Rsedaikin) [reply skiko/src/jvmMain/cpp/windows/drawlayer.cc]: Done.

File: skiko/src/awtMain/kotlin/org/jetbrains/skiko/SkiaLayer.awt.kt (hunk 4 above)
Comment (Rsedaikin) [reply skiko/src/awtMain/kotlin/org/jetbrains/skiko/SkiaLayer.awt.kt]: Done.

File: skiko/src/jvmMain/cpp/linux/drawlayer.cc [hunk 5]
```diff
@@ -6,13 +6,95 @@
...
+#include <dlfcn.h>
 #include <unistd.h>
 #include <stdio.h>
-#include <vector>
 #include "jni_helpers.h"
+#include <vector>
 
 typedef GLXContext (*glXCreateContextAttribsARBProc)(Display *, GLXFBConfig, GLXContext, Bool, const int *);
 
+static void* loadXrandr() {
```
Comment (olonho) [inline skiko/src/jvmMain/cpp/linux/drawlayer.cc:17]: Guess we may share the code by moving xrandr manipulations to its own file.

Comment (Rsedaikin) [reply skiko/src/jvmMain/cpp/linux/drawlayer.cc]: Done.

File: skiko/src/awtMain/kotlin/org/jetbrains/skiko/SkiaLayer.awt.kt (hunk 4 above)
Comment (Rsedaikin) [reply skiko/src/awtMain/kotlin/org/jetbrains/skiko/SkiaLayer.awt.kt]: Done.

**Review by olonho — APPROVED** [review]

**Pull request merged**